src/pipeline.ts

```diff
@@ -482,14 +523,400 @@ export async function scanRssFeeds() {
+		});
+		return;
+	}
+	const dirContents = await findAllTorrentFilesInDir(targetDir);
+	if (dirContents.length === 0) {
+		logger.info({
+			label: Label.INJECT,
+			message: `No torrent files found to inject in ${targetDirLog}`,
+		});
+		return;
+	}
+	logger.info({
+		label: Label.INJECT,
+		message: `Found ${chalk.bold.white(dirContents.length)} torrent file(s) to inject in ${targetDirLog}`,
+	});
+
+	const { searchees } = await findSearchableTorrents(Label.INJECT, {
+		useFilters: false,
+	});
+
+	const toDelete = new Set<string>();
+	const toRecheck = new Set<string>();
+
+	// Usually source got deleted or partial injection never completes
+	function shouldCleanUpTorrent(torrentFilePath: string) {
+		return fs.statSync(torrentFilePath).mtimeMs < Date.now() - ms("1 week");
+	}
+
+	let totalInjected = 0;
+	let totalFullMatches = 0;
+	let totalPartialMatches = 0;
+	let totalBlocked = 0;
+	let totalAlreadyExists = 0;
+	let totalCandidateIncomplete = 0;
+	let totalSearcheeIncomplete = 0;
+	let totalFailed = 0;
+	let totalUnmatched = 0;
+	let count = 0;
+	let foundBadFormat = false;
+	for (const torrentFilePath of dirContents) {
+		const progress = chalk.blue(`(${++count}/${dirContents.length})`);
+		const filePathLog = chalk.bold.magenta(torrentFilePath);
+		if (shouldCleanUpTorrent(torrentFilePath)) {
+			toDelete.add(torrentFilePath);
+		}
+		let meta: Metafile;
+		try {
+			meta = await parseTorrentFromFilename(torrentFilePath);
+		} catch (e) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${progress} Failed to parse ${filePathLog}`,
+			});
+			logger.debug(e);
+			continue;
+		}
+		const metaLog = getLogString(meta, chalk.bold.white);
+
+		const torrentNameInfo =
+			await parseInfoFromSavedTorrent(torrentFilePath);
+		const tracker = torrentNameInfo?.tracker ?? UNKNOWN_TRACKER;
+		if (tracker === UNKNOWN_TRACKER) {
+			foundBadFormat = true;
+		}
+
+		// Decide stage
+		const fullMatches: [
+			SearcheeWithLabel,
+			Decision.MATCH | Decision.MATCH_SIZE_ONLY,
+		][] = [];
+		const partialMatches: [SearcheeWithLabel, Decision.MATCH_PARTIAL][] =
+			[];
+		let foundBlocked = false;
+		for (const searchee of searchees) {
+			const { decision } = await assessCandidateHelper(
+				meta,
+				searchee,
+				[],
+			);
+			if (decision === Decision.MATCH) {
+				fullMatches.unshift([searchee, decision]);
+			} else if (decision === Decision.MATCH_SIZE_ONLY) {
+				fullMatches.push([searchee, decision]);
+			} else if (decision === Decision.MATCH_PARTIAL) {
+				partialMatches.push([searchee, decision]);
+			} else if (decision === Decision.BLOCKED_RELEASE) {
+				foundBlocked = true;
+			}
+		}
+		fullMatches.sort((a, b) => {
+			// Prefer torrent over data/virtual
+			if (a[0].infoHash && !b[0].infoHash) {
+				return -1;
+			}
+			if (!a[0].infoHash && b[0].infoHash) {
+				return 1;
+			}
+			return 0; // Should keep MATCH first within a searchee type
```

Comment:
nit: with tuples, I think it's easier to read when we use destructuring to access its members rather than [0] or [1]

```suggestion
		fullMatches.sort(([searcheeA], [searcheeB]) => {
			// Prefer torrent over data/virtual
			if (searcheeA.infoHash && !searcheeB.infoHash) {
				return -1;
			}
			if (!searcheeA.infoHash && searcheeB.infoHash) {
				return 1;
			}
			return 0; // Should keep MATCH first within a searchee type
```